content/cloud-docs/agents/hooks.mdx

```diff
@@ -0,0 +1,137 @@
+---
+page_title: Hooks - Terraform Cloud Agents - Terraform Cloud and Terraform Enterprise
+---
+
+# Hooks
+
+Terraform Cloud Agents support executing custom programs, or hooks, during
+strategic points of a Terraform run. These hooks allow operators to extend the
+functionality of Terraform runs.
+
+Terraform Cloud Agents support the following hooks:
+
+- `pre-plan` - This hook executes before `terraform init` during a `plan`
+operation.
+
+- `post-plan` - This hook executes after `terraform plan` during a `plan`
+operation regardless whether or not `terraform plan` successfully executed.
+
+- `pre-apply` - This hook executes before `terraform init` during an `apply`
+operation.
+
+- `post-apply` - This hook executes after `terraform apply` during an `apply`
+operation regardless whether or not `terraform apply` successfully executed.
+
+## Limitations
+
+Please note the following when using hooks.
+
+- If the hook exits with a non-zero exit code, the Terraform run will fail
+immediately.
+- The name of the hook is not customizable.
+- The hook must be executable. That is, the hook must have execute permission.
+- The hook has a 60 second timeout.
+- Environment variables do not persist across hooks. For example, if a
+`pre-plan` hook exports environment variables, they will not be available
+during `terraform plan` nor will they be available during the `post-plan` hook.
+Similarly, if `terraform plan` exports environment variables, they will not be
+available during the `post-plan` hook.
+
+## Configuration
+
+Hooks are stored within the `hooks` subdirectory of the agent's data directory.
+By default, that's `~/.tfc-agent/hooks`.
+
+To enable a hook, create a file named `terraform-${HOOK}` where `${HOOK}` is
+one of the hooks listed above. Then store that file within the `hooks`
+subdirectory and make it executable. Once an agent is started and receives a
+run, it will attempt to execute your hook at the specified point during the
+Terraform run.
+
+For example, a `pre-plan` hook uses an executable file named
+`~/.tfc-agent/hooks/terraform-pre-plan`.
+
+### Configuring hooks when running an Agent as a binary
+
+When running the `tfc-agent` binary directly on a Linux instance, create the
```

Comment:
Can we turn this and the below into numbered steps? That will make it easier for folks to follow!